### DISCUSSION TITLE
Simulation reporting time

### DISCUSSION THREAD

**Discussion by KrushilM** [pyswmm/pyswmm#428] [Q&A] [answered]
I am running a 50 days simulation with five minutes of runoff steps and setting step_advance as 3600 seconds. However, while printing, it is retrieving with some additional seconds initially, which later shifted by 30 mins. I attaching some of the results below.

2020-10-20 01:00:01 
2020-10-20 02:00:04 
2020-10-20 03:00:11 

at the end

2020-11-30 20:38:50 
2020-11-30 21:38:56 
2020-11-30 22:39:02

I would appreciate any help you can provide. Thank you

**Comment by bemcdonnell** [comment, marked as answer]
@KrushilM, this one is not exactly classified as a bug. When this feature was added, we did not have a feature in the SWMM-API to “set” a SWMM step advance. So in this case, pyswmm can call SWMM to advance the simulation but SWMM was still in control of the variable time stepping and time step length.  So the way we implemented step advance at that time only allowed us to step forward like:

Sum_of_variable_routing_step(s)<= step_advance < Sum_of_variable_routing_step(s)+step_advance

so if your model was stepping on a 6 second variable step and you used step_advance 32, pyswmm would have control again in 36 seconds. 

Fun fact, the latest version of swmm does now allow us to provide an actual step value. Technically this will change results of a model if it it used! However, we are going to offer this feature in the next release of pyswmm-2.x.

**Reply by KrushilM** [reply]
@bemcdonnell, Thank you for your prompt reply. Based on your explanation, I ran the model with small routing steps to get values at a particular timestep (example - 5 sec, 1 sec, 0.5 sec) and It worked. Thanks again for your help.
